hibernate-core/src/test/java/org/hibernate/test/dialect/function/TemplateRendererTest.java

```diff
@@ -0,0 +1,37 @@
+package org.hibernate.test.dialect.function;
+
+import java.util.Collections;
+
+import org.hibernate.dialect.function.TemplateRenderer;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TemplateRendererTest extends BaseCoreFunctionalTestCase {
+	
+	@Test
+	@TestForIssue( jiraKey = "HHH-13846" )
+	public void testParamIndexWithoutDigit() {
+		String template = "REGEXP_REPLACE( "
+				+ "    REPLACE( "
+				+ "        TRANSLATE( "
+				+ "            LOWER(?1), "
+				+ "            'áàãâäéèêëíìîïóòõôöúùûüç', "
+				+ "            'aaaaaeeeeiiiiooooouuuuc' "
+				+ "        ), "
+				+ "        ' ', "
+				+ "        '' "
+				+ "    ), "
+				+ "    '[\\u2000-\\u206F\\u2E00-\\u2E7F\\\\''!\"#$%&()*+,\\-.\\/:;<=>?@\\[\\]^_`{|}~]', " // -- '?'' causes the problem
```

Comment:
if the above '?' is followed by digit, it should not be treated as param, so I doubt whether the fixing is worthwhile in the first place.